routes/user.js

```diff
@@ -43,30 +44,34 @@ router.post("/add", (req, res) => {
 router.get("/verify", (req, res) => {
   const email = gUnverifiedEmails.get(req.query.state);
   if (email) {
-    gEmails.add(email);
-    gUnverifiedEmails.delete(req.query.state);
-    res.json({ email, info: `Successfully added ${email}`});
+    Subscribers.addUser(email).then(() => {
+      gUnverifiedEmails.delete(req.query.state);
+      res.json({ email, info: `Successfully added ${email}`});
+    });
```

Comment:
This needs a catch. In its current form, if addUser fails, the HTTP response is never sent.